### ISSUE TITLE
pytest-mypy only checks test files when providing testpaths

### ISSUE THREAD

**Issue by tkruse** [realpython/pytest-mypy#70]
With a src layout for files, and this setup.cfg:
```
[options]
package_dir=
  = src
packages = find:
setup_requires =
  pytest-runner

[options.packages.find]
where = src/
exclude =
    test
[tool:pytest]
python_files = *.py
# providing testpaths makes pytest-mypy only check tests
testpaths = test
addopts =
  --mypy
[mypy]
# this line sadly gets ignored using pytest-mypy
files = src
```

mypy is only applied to the files in test folder (fails on imports from src folder). This might even be following the spec of mypy (which says setup.cfg "files" will only be used if no other argument is provided), but it makes pytest-pylint less useful.

**Comment by dmtucker** [comment]
This is essentially a duplicate of #67.

tl;dr:
> [Using the Mypy `files` setting with `pytest-mypy` is] a pretty wonky thing conceptually. In some sense, the point of this plugin is to let pytest handle collection 🤔

`testpaths` affects Pytest collection, which can definitely affect what gets tested with `mypy`, but it doesn't affect the behavior of `files`.

You nailed the reason `files` gets ignored: `pytest-mypy` passes all collected `.py` files to `mypy`, so arguments are always provided.
- If no `.py` files are collected by Pytest, then execution ends after collection (i.e. there are no `MypyItem`s to run, so there is nothing to even invoke `mypy`).
- Even if `files` wasn't ignored, a file that wasn't collected by Pytest would have no corresponding `MypyItem` with which to report the outcome of the `mypy` check.

Please feel free to chime in over there if you have thoughts to share 😊
